packages/ui/src/components/va-file-upload/VaFileUploadUndo/VaFileUploadUndo.vue

```diff
@@ -1,45 +1,113 @@
 <template>
-  <div class="va-file-upload-undo">
-    <span>File was successfully deleted</span>
+  <div
+    class="va-file-upload-undo"
+    :class="{ 'va-file-upload-undo--vertical': $props.vertical }"
+  >
+    <va-progress-bar
+      :model-value="progress"
+      :rounded="false"
+      class="va-file-upload-undo__progress-bar"
+    />
+    <span class="va-file-upload-undo__text">{{ deletedFileMessage }}</span>
     <va-button
       class="va-file-upload-undo__button"
       size="small"
       outline
       @click="$emit('recover')"
     >
-      Undo
+      {{ undoButtonText }}
     </va-button>
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent } from 'vue'
+import { defineComponent, computed, inject, ref, onMounted } from 'vue'
 
-import { VaButton } from '../../va-button'
+import { VaFileUploadKey } from '../types'
+
+import { VaButton, VaProgressBar } from '../../index'
 
 export default defineComponent({
   name: 'VaFileUploadUndo',
+
   components: {
+    VaProgressBar,
     VaButton,
   },
+
+  props: {
+    vertical: { type: Boolean, default: false },
+  },
+
   emits: ['recover'],
+
+  setup: () => {
+    const progress = ref(100)
+    let timer: ReturnType<typeof setInterval>
+
+    const { undoDuration, deletedFileMessage, undoButtonText } = inject(VaFileUploadKey, {
+      undoDuration: computed(() => 3000),
+      deletedFileMessage: computed(() => 'File was successfully deleted'),
+      undoButtonText: computed(() => 'Undo'),
+    })
+
+    onMounted(() => {
+      timer = setInterval(() => {
+        progress.value -= 10
+
+        if (progress.value <= 0) {
+          progress.value = 0
+          clearInterval(timer)
+        }
+      }, undoDuration.value * 0.1)
```

Comment:
Let's make it 60fps. Or 24 a least... (1000 / 24 = 41 instead of 16)

```suggestion
      timer = setInterval(() => {
        progress.value -= undoDuration.value / 1000 * 0.16

        if (progress.value <= 0) {
          progress.value = 0
          clearInterval(timer)
        }
      }, 16)
```